### PR TITLE
Make toolchain-desktop-macos depend on fetch-nss-artifact [ci full]

### PR DESCRIPTION
**Pull request by ahal**
### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [x] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by ahal** [comment]
See commit message for explanation of what's going on here.

File: libs/build-nss-desktop.sh [hunk 1]
```diff
@@ -57,12 +57,8 @@ fi
...
 # https://github.com/mozilla/application-services/issues/962
 if [[ "${CROSS_COMPILE_TARGET}" =~ "darwin" ]]; then
-  #From https://firefox-ci-tc.services.mozilla.com/tasks/index/app-services.cache.level-3.content.v1.nss-artifact/latest
-  curl -sfSL --retry 5 --retry-delay 10 -O "https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/app-services.cache.level-3.content.v1.nss-artifact.latest/artifacts/public/dist.tar.bz2"
-  SHA256="15e8cab63a4e0484564231a24d23d8be349217ba6add53b56609c339bb306829"
-  echo "${SHA256}  dist.tar.bz2" | shasum -a 256 -c - || exit 2
-  tar xvjf dist.tar.bz2 && rm -rf dist.tar.bz2
-  NSS_DIST_DIR=$(abspath "dist")
+  # From https://firefox-ci-tc.services.mozilla.com/tasks/index/app-services.cache.level-3.content.v1.nss-artifact/latest
+  NSS_DIST_DIR="$MOZ_FETCHES_DIR/dist"
```
Comment (bendk) [inline libs/build-nss-desktop.sh:61]: This also gets run outside of CI by devs who want to build locally.  Could this be inside an if branch?  I think there's an env variable named `CI` that could be use, if not `MOZ_FETCHES_DIR` seems like a fine test.

**Review by bendk — DISMISSED** [review]
This is awesome!  The only change needed is to make sure it continues to work as before when run locally.

**Review by skhamis — APPROVED** [review]
Amazing work! Truly appreciate helping improve this!!

**Review by bendk — APPROVED** [review]

**Pull request merged**